cvat/apps/engine/backup.py

```diff
@@ -398,27 +398,31 @@ def serialize_task():
             return task
 
         def serialize_segment(db_segment):
-            db_job = db_segment.job_set.first()
-            job_serializer = SimpleJobSerializer(db_job)
-            for field in ('url', 'assignee'):
-                job_serializer.fields.pop(field)
-            job_data = self._prepare_job_meta(job_serializer.data)
+            segments = []
+            db_jobs = db_segment.job_set.all()
+            for db_job in db_jobs:
+                job_serializer = SimpleJobSerializer(db_job)
+                for field in ('url', 'assignee'):
+                    job_serializer.fields.pop(field)
+                job_data = self._prepare_job_meta(job_serializer.data)
 
-            segment_serializer = SegmentSerializer(db_segment)
-            segment_serializer.fields.pop('jobs')
-            segment = segment_serializer.data
-            segment_type = segment.pop("type")
-            segment.update(job_data)
+                segment_serializer = SegmentSerializer(db_segment)
+                segment_serializer.fields.pop('jobs')
+                segment = segment_serializer.data
+                segment_type = segment.pop("type")
+                segment.update(job_data)
 
-            if self._db_task.segment_size == 0 and segment_type == models.SegmentType.RANGE:
-                segment.update(serialize_custom_file_mapping(db_segment))
+                if self._db_task.segment_size == 0 and segment_type == models.SegmentType.RANGE:
+                    segment.update(serialize_custom_file_mapping(db_segment))
 
-            return segment
+                segments.append(segment)
+
+            return segments
```

Comment:
Have you tried to run merging on tasks restored from backups? I'm not sure it's working.